contribs/commercialTrafficApplications/src/main/java/org/matsim/contrib/commercialTrafficApplications/jointDemand/commercialJob/DefaultCommercialJobGenerator.java

```diff
@@ -291,108 +329,158 @@ private Plan createPlainPlanFromTour(Carrier carrier, ScheduledTour scheduledTou
+						carrier.getServices().put(serviceId, service);
+					} else {
+						throw new RuntimeException("Carrier Id does not exist: " + carrierId.toString()
+								+ ". There is a wrong reference in activity attribute of person " + customer);
+					}
+				}
+			}
+		}
 
-    public void removeFreightAgents(Scenario scenario) {
-        freightDrivers.forEach(d -> scenario.getPopulation().removePerson(d));
-        freightVehicles.forEach(vehicleId -> scenario.getVehicles().removeVehicle(vehicleId));
-        CarrierVehicleTypes.getVehicleTypes(carriers).getVehicleTypes().keySet().forEach(vehicleTypeId -> scenario.getVehicles().removeVehicleType(vehicleTypeId));
-    }
+	}
+
+	private void buildTours() {
+		try {
+			TourPlanning.runTourPlanningForCarriersWithNetBasedCosts(carriers, scenario, timeSliceWidth, carTT);
+		} catch (ExecutionException | InterruptedException e) {
+			e.printStackTrace();
+		}
+	}
+
+	private static Id<CarrierService> createCarrierServiceIdXForCustomer(Person customer, int x) {
+		return Id.create(customer.getId().toString() + "_" + x, CarrierService.class);
+	}
+
+	private Person createDriverPerson(Id<Person> driverId) {
+		return PopulationUtils.getFactory().createPerson(driverId);
+	}
+
+	@Override
+	public void notifyAfterMobsim(AfterMobsimEvent event) {
+
+		removeFreightAgents(false);
+		// Toggles ChangeCommercialJobOperatorStrategy to enable or disable via weight
+		// zero
+		toggleChangeCommercialJobOperatorStrategy(event.getIteration());
+
+	}
+
+	private boolean isChangeCommercialJobOperatorStrategy(GenericPlanStrategy<Plan, Person> strategy) {
+		String name = strategy.toString();
+		
+		return name.endsWith(ChangeCommercialJobOperator.SELECTOR_NAME);
+	}
+
+	private void toggleChangeCommercialJobOperatorStrategy(int currentIteration) {
+
+		// 0 means always tour planning
+		if (this.keepUsedOperatorsInterval != 0) {
+						
+			this.enableTourPlanning = ((currentIteration + 1.0) % this.keepUsedOperatorsInterval == 0.0) ? true : false;
+			
+			if (enableTourPlanning) {
+				log.info("Toggle " + ChangeCommercialJobOperator.SELECTOR_NAME);
+			}
+			
+			Collection<StrategySettings> allStrategies = this.scenario.getConfig().strategy().getStrategySettings();
+
+			for (StrategySettings strategy : allStrategies) {
+				if (strategy.getStrategyName().equals(ChangeCommercialJobOperator.SELECTOR_NAME)) {
+					double initalWeight = strategy.getWeight();
+					
+					// Update weight 
+					double newWeight = enableTourPlanning ? initalWeight : 0.0;
+					for (GenericPlanStrategy<Plan, Person> currentStrategy : this.strategyManager.getStrategies(null)) {
+						String strategyName = currentStrategy.toString();
+						if (isChangeCommercialJobOperatorStrategy(currentStrategy)) {
+							log.info("Set weight for " + strategyName + " --> "+ newWeight);
+							this.strategyManager.changeWeightOfStrategy(currentStrategy, null, newWeight);
```

Comment:
here you could insert a break statement (once the target strategy was found and adjusted)